src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -167,12 +169,36 @@ public String getServiceClientConstructorName(Interface service) {
   /**
    * Returns the local name used in Go files for the package of the proto.
    */
-  private static String localPackageName(ProtoElement proto) {
+  private String localPackageName(ProtoElement proto) {
     String goPackage = proto.getFile().getProto().getOptions().getGoPackage();
-    if (goPackage.startsWith(CORE_PROTO_BASE)) {
-      return goPackage.substring(CORE_PROTO_BASE.length()).replace("/", "_");
+    if (!goPackage.startsWith(CORE_PROTO_BASE)) {
+      return proto.getFile().getProto().getPackage().replace(".", "_");
     }
-    return proto.getFile().getProto().getPackage().replace(".", "_");
+    // Inside CORE_PROTO_BASE, we are referencing code generated by curated protos.
+    // Import path of these protos might be versioned
+    //   google.golang.org/genproto/googleapis/example/library/v1
+    // or not
+    //   google.golang.org/genproto/googleapis/api/monitoredres
+    // We heuristically get the import name by looking for the right-most element,
+    // that is not a version number.
+    List<String> parts = Arrays.asList(goPackage.split("/"));
+    Collections.reverse(parts);
+    String name = parts.get(0);
+    for (String part : parts) {
+      if (part.length() < 2 || part.charAt(0) != 'v' || !Character.isDigit(part.charAt(1))) {
```

Comment:
We already are right? `v1beta2` has length greater than or equal to 2, starts with "v", and its second character "1" is a digit. It fails all 3 tests.